src/libsyntax/ast.rs

```diff
@@ -160,7 +155,13 @@ pub struct PathSegment {
     /// this is more than just simple syntactic sugar; the use of
     /// parens affects the region binding rules, so we preserve the
     /// distinction.
-    pub parameters: PathParameters,
+    pub parameters: Option<P<PathParameters>>,
```

Comment:
Could you add a comment telling that this is a size optimization and what `None` means?